### DISCUSSION TITLE
HTTP support for the local-tx-submission protocol

### DISCUSSION THREAD

**Discussion by KtorZ** [CardanoSolutions/ogmios#268] [Archives]
#### Describe your idea, in simple words.

Enable routing HTTP requests to the local-tx-submission protocol. A typical request would be:

- verb: `POST` 
- content-type: `application/cbor` or `text/plain` 
- accept-encoding: `application/json`
- body: raw binary (when `application/cbor`) or an encoded base16 string when `text/plain`

The response would be the full local-tx-submission response, including detailed errors on failure. 

#### Why is it a good idea?

The local-tx-submission is a simple-enough protocol with no state; it is therefore a likely candidate for HTTP rest-like interfaces. It also reduces the entry barrier even more, as connecting and sending through a websocket is less idiomatic than firing an HTTP request for most ecosystem. 

#### Are you willing to work on it yourself?

Yes.

**Comment by rhyslbw** [comment]
Added in https://github.com/CardanoSolutions/ogmios/releases/tag/v6.0.0
